### PR TITLE
Initial pass at bazel build gen

### PR DESCRIPTION
**Pull request by ZacSweers**
This is a first pass at supporting bazel build file generation.

**Supported features**
- kotlin jvm targets
- kotlin jvm test targets
- kotlin android targets
- a few ksp processors we use
- a few compiler plugins we use
- a `maven_dependencies.bzl` file generation task based on the available version catalogs.

**Usage**
- Generate build files: `./gradlew generateBazel -Psgp.bazel.enable=true`
- Regenerate `maven_dependencies.bzl`: `./gradlew generateBazelMavenDependencies -Psgp.bazel.enable=true`

**Next steps**
- kotlin android test targets. It appears that every member of the bazel community just uses their own fork, which is... not helpful. This includes some toe-holds for that though, with robolectric support too.
- anvil support
- figure out how to make bazel's tests automatically run all test sources found (currently it seems to require specifying one test class)

For external consumers - this isn't something turnkey you can use in your own projects and not worth reading into at all.

### CODE REVIEW

File: slack-plugin/src/main/kotlin/slack/gradle/bazel/JvmProjectSpec.kt [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.UntrackedTask
+import org.gradle.internal.component.external.model.ModuleComponentArtifactIdentifier
+import slack.gradle.register
+
+// TODO what about java files? What does the kt_jvm_library rule do?
```
Comment (erikkerber) [inline slack-plugin/src/main/kotlin/slack/gradle/bazel/JvmProjectSpec.kt:24]: srcs supports both Kotlin and Java files

https://github.com/bazelbuild/rules_kotlin/blob/master/docs/kotlin.md#kt_jvm_library-srcs

File: slack-plugin/src/main/kotlin/slack/gradle/bazel/GenerateMavenDependenciesTask.kt [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.UntrackedTask
+import slack.gradle.findByType
+import slack.gradle.register
+
+@UntrackedTask(because = "This is an on-demand task")
+internal abstract class GenerateMavenDependenciesTask : DefaultTask() {
```
[the remark this answers is not in the document]
Comment (ZacSweers) [reply slack-plugin/src/main/kotlin/slack/gradle/bazel/GenerateMavenDependenciesTask.kt:31]: No, tbh there's no solution for that in bazel. This task wouldn't really be able to help

[the remark this answers is not in the document]
Comment (ZacSweers) [reply slack-plugin/src/main/kotlin/slack/gradle/bazel/GenerateMavenDependenciesTask.kt:31]: I think it's better to just leave that internal. I'll probably remove this task

[the remark this answers is not in the document]
Comment (ZacSweers) [reply slack-plugin/src/main/kotlin/slack/gradle/bazel/GenerateMavenDependenciesTask.kt]: Added them via `suffixMappings`

**Pull request closed without merging**